### ISSUE TITLE
use "integralGains" for single pulse extraction?

### ISSUE THREAD

**Issue by dneise** [fact-project/fact-tools#197]
When extracting the `photoncharge` we use these individual pixel gain number:
https://github.com/fact-project/fact-tools/blob/master/src/main/resources/default/gain_sorted_20131127.csv
(btw. this is no CSV at all)
They look like this:
![integral_gains_from_file](https://cloud.githubusercontent.com/assets/8200858/22109521/63ba52b6-de58-11e6-822d-62b7f80c8a61.png)

I was wondering if we should also use these numbers in the single pulse extractor, here:
https://github.com/fact-project/fact-tools/blob/master/src/main/java/fact/extraction/SinglePulseExtraction.java#L117

At the moment the constant `factSinglePeAmplitudeInMv` which is used here, is 10.
but we can also convert these integralGains from the file into this number.
The distribution for all pixels looks like this then:
![conversion_factors](https://cloud.githubusercontent.com/assets/8200858/22109410/e836dcae-de57-11e6-8661-4cd0b738c706.png)

**Comment by jebuss** [comment]
Good idea, makes sense to me. Let's test it. However the amplitude might only be a 10% effect at max.

**Comment by dneise** [comment]
Is it a good idea then? Or a waste of time? We have very very limit amout of time only. 
The single photon extraction pass should have been started since long. 

Question is not whether its nice or not ... surely one can do it and test it ... but is it necessary? 
Will people say: "This extraction is invalid", if we do not use these individual gains?

**Comment by jebuss** [comment]
I don't know if it is necessary or not. But I think it would not make the result worse, only better! That's my guess. So put it in, if you agree. If we don't use it, we might misestimate Cherenkov pulses by up to 10%, since we apply a 10mv Amplitude to a 9mv pulse, which is also my guess.

**Comment by dneise** [comment]
> since we apply a 10mv Amplitude to a 9mv pulse, which is also my guess.

Have a look at the lower image: 
https://cloud.githubusercontent.com/assets/8200858/22109410/e836dcae-de57-11e6-8661-4cd0b738c706.png

It shows the factors, which would be applied instead of the constant factor 10. ...

**Comment by jebuss** [comment]
Yes that's what I meant.

**Comment by jebuss** [comment]
I was talking about the worst case which is the outlier at 9mv

**Comment by jebuss** [comment]
To shorten the discussion. I think it's a good idea. I think it's worth it. I thinking would improve the result. I think we should go for it.

**Comment by dneise** [comment]
Adrian agreed with you

**Comment by relleums** [comment]
I introduced a single pulse gain calibration service to avoid code duplication. However, I needed to modify ALL xmls.

**Comment by MaxNoe** [comment]
Related to #267

**Comment by MaxNoe** [comment]
The GainService is now in the master
